.github/workflows/lint-test.yml

```diff
@@ -0,0 +1,65 @@
+name: Lint and Test
+
+on:
+  push:
+    branches-ignore:
+      - main
+  workflow_dispatch: {}
+
+env:
+  RAILS_ENV: test
+  CM_DB_NAME: centralized_metadata_test
+  CM_DB_PASSWORD: postgres
+
+jobs:
+  test:
+    runs-on: ubuntu-latest
+
+    strategy:
+      matrix:
+        ruby-version: ['3.4']
+
+    services:
```

Comment:
@cdoyle-temple Please look at what we are doing in other actions.  I don't think we need all of this. TU Press would be a good example.